lib/submissions.ts

```diff
@@ -0,0 +1,73 @@
+import axios from 'axios';
+import { Submission, StepAnswers } from 'data/flexibleForms/forms.types';
+
+const { ENDPOINT_API, AWS_KEY } = process.env;
+
+const headersWithKey = {
+  'x-api-key': AWS_KEY,
+};
+
+export const startSubmission = async (
+  formId: string,
+  socialCareId: number,
+  createdBy: string
+): Promise<Submission> => {
+  const { data } = await axios.post(
+    `${ENDPOINT_API}/submissions`,
+    {
+      formId,
+      socialCareId,
+      createdBy,
+    },
+    {
+      headers: headersWithKey,
+    }
+  );
+  return data;
+};
+
+export const getSubmissionById = async (
+  submissionId: string
+): Promise<Submission> => {
+  const { data } = await axios.get(
+    `${ENDPOINT_API}/submissions/${submissionId}`,
+    {
+      headers: headersWithKey,
+    }
+  );
+  return data;
+};
+
+export const patchSubmissionForStep = async (
+  submissionId: string,
+  stepId: string,
+  editedBy: string,
+  stepAnswers: StepAnswers
+): Promise<Submission> => {
+  const { data } = await axios.patch(
+    `${ENDPOINT_API}/submissions/${submissionId}/steps/${stepId}`,
```

Comment:
this seems the most sensible way to provide the `stepId` - it feels confusing to put it in the body because it doesn't end up in the stored record.

alternatively we could use a query parameter: `/submissions/123?step=456`

the stepId itself could be any arbitrary value, and will differ for every form